lib/my_app.rb

```diff
@@ -0,0 +1,151 @@
+ENV['RACK_ENV'] ||= "development"
+
+require_relative '../data_mapper_setup.rb'
+require 'sinatra/base'
+require 'sinatra/flash'
+
+class MakersBnb < Sinatra::Base
+  use Rack::MethodOverride
+  register Sinatra::Flash
+
+  enable :sessions
+  set :session_secret, 'super secret'
+
+  helpers do
+    def current_user
+      @current_user ||= User.get(session[:user_id]) 
+    end
+  end
+
+  get '/' do
+    @spaces = Space.all
+    erb :index
+  end
+
+  post '/' do
+    a_space = Space.create(params)  
+    current_user.space << a_space
+    current_user.save
```

Comment:
I'd prefer this in a model method. `user.add_space(space)`. Then define `add_space(space)` on the `User` model.